lib/rbi/rewriters/replace_attributes_with_methods.rb

```diff
@@ -0,0 +1,147 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    class ReplaceAttributesWithMethods < Visitor
+      extend T::Sig
+
+      sig { override.params(node: T.nilable(Node)).void }
+      def visit(node)
+        return unless node
+
+        case node
+        when Tree
+          node.nodes.dup.each do |child|
+            visit(child)
+            next unless child.is_a?(Attr)
+
+            child.replace_with_multiple(child.convert_to_methods)
+            child.detach
+          end
+        end
+      end
+    end
+  end
+
+  class AttrReader
+    sig { returns(T::Array[Method]) }
+    def convert_to_methods
+      raise "How can there be multiple?" unless sigs.one?
+      raise "TODO: Support one attr decl with multiple names" unless names.one?
+
+      name = names.first.to_s
+
+      m = Method.new(
+        name,
+        params: [],
+        visibility: visibility,
+        sigs: sigs,
+        loc: loc,
+        comments: comments,
+      )
+
+      [m]
+    end
+  end
+
+  class AttrWriter
+    sig { returns(T::Array[Method]) }
+    def convert_to_methods
+      raise "How can there be multiple?" unless sigs.one?
```

Comment:
Not for attributes, though. [Sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7B%20returns%28Integer%29%20%7D%0Asig%20%7B%20returns%28Integer%29%20%7D%0Aattr_reader%20%3Afoo%0A)

Reply:
In `.rbi` files only. With sorbet.run you're testing with a `.rb` file.

Reply:
Indeed, looks like it allows it, but the revealed type is whichever one you listed first. I think we should warn in that case, it likely wasn't intended.

Reply:
We currently do not do any linting / warning inside `rbi`, it's up to the client to make sure that what is produced is correct.

This could be a warning in Tapioca when pulling the gem's hand-written annotations. Or it could be done by Sorbet.